### PR TITLE
Feat: Loans module

### PR DESCRIPTION
**Pull request by peterslany**
Merging loans branch back into master with updated runtime and client versions.

### CODE REVIEW

File: src/parachain/loans.ts [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+     * @returns Array of borrow positions of account.
+     */
+    getBorrowPositionsOfAccount(accountId: AccountId): Promise<Array<BorrowPosition>>;
+
+    /**
+     * Get all loan assets.
+     *
+     * @returns Array of all assets that can be lent and borrowed.
+     * @remarks Method could be refactored to compute APR in lib if we can get underlyingCurrency/rewardCurrency exchange rate,
+     * but is it safe to assume that exchange rate for btc/underlyingCurrency will be
```
Comment (savudani8) [inline src/parachain/loans.ts:46]: it is safe to assume that, yes

File: src/parachain/loans.ts [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+    /**
+     * Same as `withdraw`, but exits full position.
+     *
+     * @param underlyingCurrency Currency to fully withdraw.
+     */
+    withdrawAll(underlyingCurrency: CurrencyExt): Promise<void>;
+
+    /**
+     * Enable lend position of account as collateral for borrowing.
+     *        await this._checkMarketState(underlyingCurrency, "withdraw");
```
Comment (savudani8) [inline src/parachain/loans.ts:107]: leftover code here?

File: src/parachain/loans.ts [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+            this.api.query.loans.accountDeposits(lendTokenId, accountId),
+            this.api.rpc.loans.getMarketStatus(underlyingCurrencyId),
+        ]);
+
+        const startingExchangeRate = decodeFixedPointType(accountEarned.exchangeRatePrior);
+        const currentExchangeRate = decodeFixedPointType(currentMarketStatus[2]);
+        const earnedPrior = decodeFixedPointType(accountEarned.totalEarnedPrior);
+        const earnedInterest = currentExchangeRate
+            .sub(startingExchangeRate)
+            .mul(underlyingCurrencyAmount) // TODO: check whether to use underlying or lendtoken balance here
```
Comment (savudani8) [inline src/parachain/loans.ts:271]: This should be the lendtoken

**Review by savudani8 — APPROVED** [review]

File: src/parachain/loans.ts (hunk 2 above)
Comment (peterslany) [reply src/parachain/loans.ts:107]: Yes, probably pasted there by mistake.

File: src/parachain/loans.ts (hunk 3 above)
Comment (peterslany) [reply src/parachain/loans.ts:271]: Thanks! I updated the computation :)

File: src/parachain/loans.ts (hunk 1 above)
Comment (peterslany) [reply src/parachain/loans.ts:46]: Good to know! This can be refactored later if there is a need - for now I think the current solution works well enough.

**Pull request merged**